cups/ppd-cache.c

```diff
@@ -1419,12 +1436,41 @@ _ppdCacheCreateWithPPD(ppd_file_t *ppd)	/* I - PPD file */
 
       map->pwg = strdup(pwg_name);
       map->ppd = strdup(choice->choice);
+    }
+
+   /*
+    * Since three PPD name patterns can map to "auto", their match counts
+    * should each be the count of all three combined.
+    */
+
+    match_counts[0] = match_counts[1] = match_counts[2] =
```

Comment:
I'm not sure this evaluates in the order you think it does. Need to assign the total to a temporary variable (i or j is fine to use here) and *then* assign to each of them...

Reply:
I'm pretty sure it does evaluate in the order I think it does? Addition has higher precedence than assignment, so in a chained assignment like this, the sum is evaluated first, then that value is assigned to `match_counts[2]`, then to `[1]`, then to `[0]`.

I can still use a temporary variable if you think it's clearer that way, though.

Reply:
I believe the fact that you are adding and assigning multiple variables introduces undefined behavior in C, so I would prefer to see the three elements totaled and then assigned.

Reply:
Changed it to use a temporary variable now.